### PR TITLE
chore: Update label syntax for analytics metadata

### PR DESCRIPTION
**Pull request by fralongo**
### Description

See https://github.com/cloudscape-design/components/pull/2502#discussion_r1689172124

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/button-dropdown/analytics-metadata/interfaces.ts
Comment (just-boris) [inline src/button-dropdown/analytics-metadata/interfaces.ts:4]: This one is not compatible with `exports` declaration in the toolkit package

Comment (fralongo) [reply src/button-dropdown/analytics-metadata/interfaces.ts:4]: I have already introduced it in other places, will fix the export, but won't manage to propagate it in time

Comment (fralongo) [reply src/button-dropdown/analytics-metadata/interfaces.ts:4]: I would rather merge this now, and only update the import later

Comment (fralongo) [reply src/button-dropdown/analytics-metadata/interfaces.ts:4]: PR ready in component-toolkit: https://github.com/cloudscape-design/component-toolkit/pull/85

Comment (just-boris) [reply src/button-dropdown/analytics-metadata/interfaces.ts:4]: Exposing `internal/analytics-metadata/interfaces` as one more export looks like mess. I think we need to re-export all features from `internal/analytics-metadata` instead

For the timeline, if you say there are already instances of this reached live, we can merge this too.

But if not, we will have to put this on hold until the value is exported properly

Comment (fralongo) [reply src/button-dropdown/analytics-metadata/interfaces.ts:4]: https://github.com/search?q=repo%3Acloudscape-design%2Fcomponents%20%40cloudscape-design%2Fcomponent-toolkit%2Finternal%2Fanalytics-metadata%2Finterfaces&type=code

But none reached live I believe

Comment (just-boris) [reply src/button-dropdown/analytics-metadata/interfaces.ts]: Discussed in DM, this can be merged after the related changes in component-toolkit are released

**Review by just-boris — APPROVED** [review]

**Pull request merged**
